pymeasure/instruments/keithley/keithley2602B.py

```diff
@@ -0,0 +1,213 @@
+#
+
+import logging
+
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.keithley.keithley2600 import Keithley2600
+from pymeasure.instruments.validators import strict_discrete_set, truncated_range
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+class Keithley2602B(Keithley2600):
+    """Represents the Keithley 2602B SourceMeter. This class adds digital I/O
+    functionality to the Keithley2600 series driver. Note that this driver can
+    be used to control any of the other 2600 series models that support digital
+    I/O (2601B, 2611B, 2612B, 2635B, 2636B).
+    """
+
+    number_of_pins = 14
+
+    def __init__(self, adapter, **kwargs):
+        super().__init__(adapter, includeSCPI=False, **kwargs)
+
+        self.dio_pins = []
+        for i in range(1, Keithley2602B.number_of_pins + 1):
+            self.dio_pins[i] = DigitalIOPin(self, i)
+
+    @staticmethod
+    def get_trigger_event_description_strings():
+        """Returns a list of the valid event IDs that can be used to select the event that
+        causes a trigger to be asserted on the digital output line. The list can be indexed
+        to set a digitial I/O line to assert a trigger given the described conditions. E.g.
+        to set digital line 4 to asser a trigger when the SMU completes a source
+        action on channel A,
+        use the following:
+
+        .. code-block:: python
+
+            #Assume a Keithley2602B object called "smu" has been successully instaniated
+            trigger_event_lists = smu.get_trigger_event_description_strings()
+            smu.dio_pins[3].stimulus = trigger_event_lists[0]
+
+        See page 9-61 of the Reference Manual for more details about the various event
+        descriptions.
+        """
+        event_descriptions = []
+
+        for channel in ["a", "b"]:
+            event_descriptions.append(f"smu{channel}.trigger.SOURCE_COMPLETE_EVENT_ID")
+            event_descriptions.append(f"smu{channel}.trigger.MEASURE_COMPLETE_EVENT_ID")
+            event_descriptions.append(f"smu{channel}.trigger.PULSE_COMPLETE_EVENT_ID")
+            event_descriptions.append(f"smu{channel}.trigger.SWEEP_COMPLETE_EVENT_ID")
+            event_descriptions.append(f"smu{channel}.trigger.IDLE_EVENT_ID")
+
+        return event_descriptions
+
+
+class DigitalIOPin:
+    def __init__(self, instrument, pin_number):
+        self.instrument = instrument
+        self.pin_number = pin_number
+
+    def ask(self, cmd):
+        return self.instrument.ask(f"print(digio.trigger[{self.pin_number}].{cmd})")
+
+    def write(self, cmd):
+        self.instrument.write(f"digio.trigger.[{self.pin_number}].{cmd}")
+
+    def check_errors(self):
+        return self.instrument.check_errors()
+
+    def assert_trigger(self):
+        """This method asserts a trigger pulse on one of the digital I/O lines."""
+
+        log.info("Asserting a trigger pulse on pin number %d." % self.pin_number)
```

Comment:
if we're able to use f-strings further up (e.g. line 72), it should be fine here, too
```suggestion
        log.info(f"Asserting a trigger pulse on pin number {self.pin_number}.")
```
alternatively if we want python 3.6 support, f-strings further up should be replaced by an older pattern.